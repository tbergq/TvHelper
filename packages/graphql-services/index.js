// @flow

export { default as fetch } from './src/fetch';
export { default as generateExecuteTestQuery } from './src/generateExecuteTestQuery';

// graphql shared types
export { default as LoginType } from './src/shared/types/output/LoginType';

// graphql shared resolvers
export { default as LoginResolver } from './src/shared/resolvers/LoginResolver';

// flow types
export type { SearchTvShowType } from './src/types/tvhelper/search/SearchApiResponse';
export type { GraphqlContextType } from './src/types/GraphqlContext';
export type { TvShow } from './src/types/tvhelper/tvshow/TvShow';
export type { Cast } from './src/types/tvhelper/tvshow/TvShow';
export type { Person } from './src/types/tvhelper/tvshow/TvShow';
export type { User } from './src/types/tvhelper/account/User';
