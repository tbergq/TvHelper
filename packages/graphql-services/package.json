{
  "name": "@tbergq/graphql-services",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "@adeira/fetch": "^1.0.0",
    "@kiwicom/graphql-global-id": "^0.13.0",
    "dataloader": "^1.4.0",
    "dotenv": "^8.2.0",
    "graphql": "^14.5.8",
    "graphql-relay": "^0.6.0",
    "jsonwebtoken": "^8.5.1",
    "password-hash": "^1.2.2"
  }
}
