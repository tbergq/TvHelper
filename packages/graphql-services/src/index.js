// @flow

export { default as fetch } from './fetch';

// graphql shared types
export { default as LoginType } from './shared/types/output/LoginType';
export { default as RangeDelete } from './shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './shared/resolvers/LoginResolver';

export type { LoggedInUser } from './shared/resolvers/LoginResolver';
export type { Apps } from './shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './connectionFromArray';
export type { ConnectionArguments, Connection } from './connectionFromArray';
export type { GraphQLConnectionDefinitions } from './types';
