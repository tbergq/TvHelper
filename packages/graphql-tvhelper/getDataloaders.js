// @flow

import { FavoritesRepository } from '@tbergq/tvhelper-persistence';

import SearchTvShowLoader from './src/search/dataloaders/SearchTvShowLoader';
import UserLoader from './src/account/dataloaders/UserLoader';
import TvDetailLoader from './src/tvshow/dataloaders/TvDetailLoader';
import EpisodesLoader from './src/episode/dataloaders/EpisodesLoader';
import FavoritesLoader from './src/tvshow/dataloaders/FavoritesLoader';

export default function getDataloaders() {
  const favoritesRepository = new FavoritesRepository();
  return {
    searchTvShow: SearchTvShowLoader(),
    user: UserLoader(),
    tvDetail: TvDetailLoader(),
    episodes: EpisodesLoader(),
    favorites: FavoritesLoader(favoritesRepository),
  };
}
