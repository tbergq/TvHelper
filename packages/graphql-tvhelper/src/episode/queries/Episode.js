// @flow

import { GraphQLNonNull, GraphQLID } from 'graphql';
import { fromGlobalId } from '@kiwicom/graphql-global-id';
import type { GraphqlContextType } from '@tbergq/graphql-services';

import Episode from '../types/output/Episode';

type Args = {|
  +id: string,
|};

export default {
  name: 'Episode',
  description: 'Load episode by id',
  type: Episode,
  args: {
    id: {
      type: GraphQLNonNull(GraphQLID),
    },
  },
  resolve: async (_: mixed, args: Args, { dataLoader }: GraphqlContextType) => {
    const id = fromGlobalId(args.id);
    const episode = await dataLoader.tvhelper.episode.load(id);
    return episode;
  },
};
