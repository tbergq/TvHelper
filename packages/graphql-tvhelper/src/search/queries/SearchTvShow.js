// @flow

import { GraphQLString, GraphQLNonNull } from 'graphql';
import { connectionFromArray, connectionArgs, type ConnectionArguments } from 'graphql-relay';
import type { GraphqlContextType, TvShow as TvShowType } from '@tbergq/graphql-services';

import TvShowConnection from '../../tvshow/types/output/TvShowConnection';

type Args = {|
  +query: string,
  ...$Exact<ConnectionArguments>,
|};

export default {
  name: 'SearchTvShow',
  description: 'Search for tv shows by name',
  type: TvShowConnection,
  args: {
    query: {
      type: GraphQLNonNull(GraphQLString),
    },
    ...connectionArgs,
  },
  resolve: async (_: mixed, args: Args, { dataLoader }: GraphqlContextType) => {
    const tvShows = await dataLoader.tvhelper.searchTvShow.load(args.query);

    return connectionFromArray<TvShowType>(tvShows, args);
  },
};
