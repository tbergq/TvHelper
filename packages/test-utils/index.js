// @flow

export { default as withMongoDb } from './src/useMongoDb';
