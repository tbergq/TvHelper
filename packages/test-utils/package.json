{
  "name": "@tbergq/test-utils",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "mongodb-memory-server": "^5.2.0",
    "mongoose": "^5.6.9"
  }
}
