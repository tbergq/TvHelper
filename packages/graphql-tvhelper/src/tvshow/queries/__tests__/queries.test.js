// @flow

import { generateTestsFromFixtures } from '@kiwicom/test-utils';
import { generateExecuteTestQuery } from '@tbergq/graphql-services';

import queries from '../../../../TvhelperQueries';
import getDataloaders from '../../../../getDataloaders';

jest.mock('@tbergq/tvhelper-persistence', () => {
  const pack = jest.requireActual('@tbergq/tvhelper-persistence');
  return {
    ...pack,
    FavoritesRepository: () => ({
      getFavorites: jest.fn(() => Promise.resolve([{ id: '123', userId: '123', serieId: '6' }])),
    }),
  };
});
const context = {
  user: {
    id: '123',
  },
  dataLoader: {
    tvhelper: getDataloaders(),
  },
};
const executeTestQuery = generateExecuteTestQuery(queries, null, context);

describe('queries', () => {
  generateTestsFromFixtures(`${__dirname}/__fixtures__`, input => executeTestQuery(input, null));
});
