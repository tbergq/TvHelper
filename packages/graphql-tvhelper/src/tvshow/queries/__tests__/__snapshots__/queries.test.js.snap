// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: tvDetail.graphql 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
query detail {
  tvShowDetail(id:"dHZTaG93OjY=") {
    id
    name
    status
    premiered
    rating
    image {
      original
    }
    summary
    withoutTags: summary(stripTags: false)
    cast {
      person {
        name
      }
      character {
        name
      }
    }
  }
}

~~~~~~~~~~ OUTPUT ~~~~~~~~~~
{
  "data": {
    "tvShowDetail": {
      "id": "VHZTaG93OjY=",
      "name": "The 100",
      "status": "Running",
      "premiered": "2014-03-19",
      "rating": 7.9,
      "image": {
        "original": "http://static.tvmaze.com/uploads/images/original_untouched/189/474423.jpg"
      },
      "summary": "Ninety-seven years ago, nuclear Armageddon decimated planet Earth, destroying civilization. The only survivors were the 400 inhabitants of 12 international space stations that were in orbit at the time. Three generations have been born in space, the survivors now number 4,000, and resources are running out on their dying \\"Ark\\" - the 12 stations now linked together and repurposed to keep the survivors alive. Draconian measures including capital punishment and population control are the order of the day, as the leaders of the Ark take ruthless steps to ensure their future, including secretly exiling a group of 100 juvenile prisoners to the Earth's surface to test whether it's habitable.",
      "withoutTags": "<p>Ninety-seven years ago, nuclear Armageddon decimated planet Earth, destroying civilization. The only survivors were the 400 inhabitants of 12 international space stations that were in orbit at the time. Three generations have been born in space, the survivors now number 4,000, and resources are running out on their dying \\"Ark\\" - the 12 stations now linked together and repurposed to keep the survivors alive. Draconian measures including capital punishment and population control are the order of the day, as the leaders of the Ark take ruthless steps to ensure their future, including secretly exiling a group of 100 juvenile prisoners to the Earth's surface to test whether it's habitable.</p>",
      "cast": [
        {
          "person": {
            "name": "Bob Morley"
          },
          "character": {
            "name": "Bellamy Blake"
          }
        },
        {
          "person": {
            "name": "Eliza Taylor-Cotter"
          },
          "character": {
            "name": "Clarke Griffin"
          }
        },
        {
          "person": {
            "name": "Marie Avgeropoulos"
          },
          "character": {
            "name": "Octavia Blake"
          }
        },
        {
          "person": {
            "name": "Lindsey Morgan"
          },
          "character": {
            "name": "Raven Reyes"
          }
        },
        {
          "person": {
            "name": "Paige Turco"
          },
          "character": {
            "name": "Abigail Griffin"
          }
        },
        {
          "person": {
            "name": "Richard Harmon"
          },
          "character": {
            "name": "Jonathan Murphy"
          }
        },
        {
          "person": {
            "name": "Henry Ian Cusick"
          },
          "character": {
            "name": "Marcus Kane"
          }
        },
        {
          "person": {
            "name": "Christopher Larkin"
          },
          "character": {
            "name": "Monty Green"
          }
        },
        {
          "person": {
            "name": "Isaiah Washington"
          },
          "character": {
            "name": "Thelonious Jaha"
          }
        },
        {
          "person": {
            "name": "Devon Bostick"
          },
          "character": {
            "name": "Jasper Jordan"
          }
        },
        {
          "person": {
            "name": "Tasya Teles"
          },
          "character": {
            "name": "Echo"
          }
        },
        {
          "person": {
            "name": "Ricky Whittle"
          },
          "character": {
            "name": "Lincoln"
          }
        },
        {
          "person": {
            "name": "Thomas McDonell"
          },
          "character": {
            "name": "Finn Collins"
          }
        },
        {
          "person": {
            "name": "Zach McGowan"
          },
          "character": {
            "name": "Roan"
          }
        },
        {
          "person": {
            "name": "Shannon Kook"
          },
          "character": {
            "name": "Jordan Jasper Green"
          }
        },
        {
          "person": {
            "name": "Eli Goree"
          },
          "character": {
            "name": "Wells Jaha"
          }
        },
        {
          "person": {
            "name": "Kelly Hu"
          },
          "character": {
            "name": "Callie Cartwig"
          }
        }
      ]
    }
  }
}
`;
